lib/_http_server.js

```diff
@@ -476,6 +476,7 @@ function onParserExecuteCommon(server, socket, parser, state, ret, d) {
   resetSocketTimeout(server, socket, state);
 
   if (ret instanceof Error) {
+    ret.rawPacket = d || parser.getCurrentBuffer();
```

Comment:
I'm not sure, but can you maybe move the

```js
     if (!d)
       d = parser.getCurrentBuffer();
```

block in front of the `if (ret instanceof Error)` part? Then you can always just use `d` inside of this function

Reply:
Refs: https://github.com/nodejs/node/pull/17672#discussion_r157119618